js/binancetr.js

```diff
@@ -0,0 +1,308 @@
+        //                ["46203.21000000","1.75479000"],
+        //                ["46203.03000000","0.06255000"],
+        //                ["46202.47000000","0.00170000"],
+        //                ["46202.46000000","0.01378000"],
+        //                ["46202.30000000","0.01634000"]
+        //            ],
+        //            "asks":[
+        //                ["46203.22000000","0.01611000"],
+        //                ["46203.24000000","0.00498000"],
+        //                ["46203.26000000","0.01136000"],
+        //                ["46205.59000000","0.00216000"],
+        //                ["46205.84000000","0.08579000"]
+        //            ]
+        //        },
+        //        "timestamp":1641383770656
+        //    }
+        //
+        const result = this.safeValue (response, 'data', {});
+        const timestamp = this.safeInteger (response, 'timestamp', {});
+        const orderBook = this.parseOrderBook (result, symbol, timestamp, 'bids', 'asks', 0, 1);
+        orderBook['nonce'] = this.safeInteger (result, 'lastUpdateId', {});
+        return orderBook;
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default: 500
+        }
+        const response = await this.publicGetTrades (this.extend (request, params));
+        //
+        //  [
+        //    {"id":1211494273,"price":"46410.42000000","qty":"0.00084000","quoteQty":"38.98475280","time":1641391959076,"isBuyerMaker":false,"isBestMatch":true},
+        //    {"id":1211494274,"price":"46410.42000000","qty":"0.00084000","quoteQty":"38.98475280","time":1641391959278,"isBuyerMaker":false,"isBestMatch":true},
+        //    {"id":1211494275,"price":"46410.41000000","qty":"0.00061000","quoteQty":"28.31035010","time":1641391959289,"isBuyerMaker":true,"isBestMatch":true},
+        //    {"id":1211494276,"price":"46410.42000000","qty":"0.00084000","quoteQty":"38.98475280","time":1641391959357,"isBuyerMaker":false,"isBestMatch":true},
+        //    {"id":1211494277,"price":"46410.42000000","qty":"0.00084000","quoteQty":"38.98475280","time":1641391959483,"isBuyerMaker":false,"isBestMatch":true},
+        //  ]
+        //
+        const data = this.safeValue (response, []);
+        return this.parseTrades (data, market, since, limit);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api] + this.version + '/' + path;
+        if (api === 'public') {
+            if (Object.keys (params).length) {
+                url += '?' + this.urlencode (params);
+            }
+        }
+        if (api === 'public' && path === 'trades') {
+            url = this.urls['api']['public3'] + path;
```

Comment:
Why do we use `public3` here and nowhere else? Looks like it's pointing to Binance, not Binancetr?

Reply:
Yes, there is a reason for this. But we can approach this a different way also if prefered.
Binancetr has different public endpoints URLs for different functions (and by the way, doesn't work like they say on their page).
So for fetchTrades I want it to use a different URL that I named public3.